lib/pf/Authentication/Source/FacebookSource.pm

```diff
@@ -25,44 +27,17 @@ has 'redirect_url' => (isa => 'Str', is => 'rw', required => 1, default => 'http
 has 'domains' => (isa => 'Str', is => 'rw', required => 1, default => '*.facebook.com,*.fbcdn.net,*.akamaihd.net');
 has 'create_local_account' => (isa => 'Str', is => 'rw', default => 'no');
 
-=head2 available_actions
+=head2 lookup_from_provider_info
 
-For an oauth2 source, we limit the available actions to B<set role>, B<set access duration>, and B<set unreg date>.
+Lookup the person information from the authentication hash received during the OAuth process
 
 =cut
 
-sub available_actions {
-    return [
-            $Actions::SET_ROLE,
-            $Actions::SET_ACCESS_DURATION,
-            $Actions::SET_UNREG_DATE,
-           ];
-}
-
-=head2 available_attributes
-
-=cut
-
-sub available_attributes {
-    my $self = shift;
-    return([@{$self->SUPER::available_attributes}, {value => 'username', type => $Conditions::SUBSTRING }]);
-}
-
-=head2 match_in_subclass
-
-=cut
+sub lookup_from_provider_info {
+    my ( $self, $pid, $info ) = @_;
+    my $logger = get_logger();
 
-sub match_in_subclass {
-    my ($self, $params, $rule, $own_conditions, $matching_conditions) = @_;
-    my $username =  $params->{'username'};
-    foreach my $condition (@{ $own_conditions }) {
-        if ($condition->{'attribute'} eq "username") {
-            if ( $condition->matches("username", $username) ) {
-                push(@{ $matching_conditions }, $condition);
-            }
-        }
-    }
-    return $username;
+    person_modify( $pid, firstname => $info->{first_name}, lastname => $info->{last_name} );
```

Comment:
I suggest to add the portal and the source_id as param (we use that to instantiate the correct portal profile on  the portal on the management interface)